rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -281,6 +284,88 @@ void getActiveFederatorBtcPublicKey_returnsExpectedFederatorBtcPublicKey() {
             byte[] activeFederatorBtcPublicKey = federationSupport.getActiveFederatorBtcPublicKey(0);
             assertThat(activeFederatorBtcPublicKey, is(newFederation.getBtcPublicKeys().get(0).getPubKey()));
         }
+
+        @Test
+        @Tag("getActiveFederatorPublicKeyOfType")
+        void getActiveFederatorPublicKeyOfType_returnsExpectedFederatorPublicKeys() {
+            BtcECKey federatorFromNewFederationBtcPublicKey = newFederation.getBtcPublicKeys().get(0);
+            byte[] activeFederatorBtcPublicKey = federationSupport.getActiveFederatorPublicKeyOfType(0, KeyType.BTC);
+            assertThat(activeFederatorBtcPublicKey, is(federatorFromNewFederationBtcPublicKey.getPubKey()));
+
+            // since new federation was created without specifying rsk public keys,
+            // these are set deriving the btc public keys
+            ECKey federatorFromNewFederationRskPublicKey = ECKey.fromPublicOnly(federatorFromNewFederationBtcPublicKey.getPubKey());
+            byte[] activeFederatorRskPublicKey = federationSupport.getActiveFederatorPublicKeyOfType(0, KeyType.RSK);
+            assertThat(activeFederatorRskPublicKey, is(federatorFromNewFederationRskPublicKey.getPubKey(true)));
+
+            // since new federation was created without specifying mst public keys,
+            // these are set copying the rsk public keys
+            byte[] activeFederatorMstPublicKey = federationSupport.getActiveFederatorPublicKeyOfType(0, KeyType.MST);
+            assertThat(activeFederatorMstPublicKey, is(federatorFromNewFederationRskPublicKey.getPubKey(true)));
+        }
+
+        @Test
+        @Tag("getActiveFederatorPublicKeyOfType")
+        void getActiveFederatorPublicKeyOfType_withSpecificRskKeys_returnsExpectedFederatorPublicKeys() {
+            // new federation with specific rsk public keys
+            List<ECKey> rskECKeys = BitcoinTestUtils.getEcKeysFromSeeds(
+                new String[]{"rsk01", "rsk02", "rsk03", "rsk04", "rsk05", "rsk06", "rsk07", "rsk08", "rsk09"}
+            );
+            newFederation = new P2shErpFederationBuilder()
+                .withMembersRskPublicKeys(rskECKeys)
+                .build();
+            storageProvider.setNewFederation(newFederation);
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .build();
+
+            BtcECKey federatorFromNewFederationBtcPublicKey = newFederation.getBtcPublicKeys().get(0);
+            byte[] activeFederatorBtcPublicKey = federationSupport.getActiveFederatorPublicKeyOfType(0, KeyType.BTC);
+            assertThat(activeFederatorBtcPublicKey, is(federatorFromNewFederationBtcPublicKey.getPubKey()));
+
+            ECKey federatorFromNewFederationRskPublicKey = rskECKeys.get(0);
```

Comment:
out of scope, but we should probably create `getRskPublicKeys()`, `getMstPublicKeys()` methods for federations.
Without them, this test is not testing 100% what we want